src-cpp/app/modules/main/view.cpp

```diff
@@ -15,5 +16,27 @@ void View::load()
     emit viewLoaded();
 }
 
+void View::addItem(Shared::Models::SectionItem* item)
+{
+    m_sectionModel.addItem(item);
+    emit sectionsModelChanged();
+}
+
+SectionModel* View::getSectionsModel()
+{
+    return &m_sectionModel;
+}
+
+SectionItem* View::getActiveSection()
+{
+    return m_sectionModel.getActiveItem();
+}
+
+void View::setActiveSection(QString Id)
+{
+    m_sectionModel.setActiveSection(Id);
+    activeSectionChanged();
```

Comment:
`QString id` should be `const QString& id`

Reply:
and i think we should check equal before emit some signals to avoid useless calls.

Reply:
Agreed! will improve :)